src/Microsoft.Health.Dicom.Core/Features/FellowOakDicom/CustomDicomImplementation.cs

```diff
@@ -0,0 +1,34 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Reflection;
+using FellowOakDicom;
+
+namespace Microsoft.Health.Dicom.Core.Features.FellowOakDicom;
+
+/// <summary>
+/// Represents a custom DICOM implementation.
+/// </summary>
+public static class CustomDicomImplementation
+{
+    /// <summary>
+    /// Azure Health Data Services specific OID registered under Microsoft OID arc.
+    /// Used to identify the DICOM implementation Class UID.
+    /// </summary>
+    private const string ImplementationClassUid = "1.3.6.1.4.1.311.129";
+
+    /// <summary>
+    /// This method sets the DICOM implementation class UID and version.
+    /// ImplementationClassUID and ImplementationVersion are used to identify the software that generated or last touched the data.
+    /// </summary>
+    public static void SetDicomImplementationClassUIDAndVersion()
+    {
+        Version version = typeof(CustomDicomImplementation).GetTypeInfo().Assembly.GetName().Version;
+
+        DicomImplementation.ClassUID = new DicomUID(ImplementationClassUid, "Implementation Class UID", DicomUidType.Unknown);
+        DicomImplementation.Version = $"{version.Major}.{version.Minor}.{version.Build}";
```

Comment:
Should this be `{Major}.{Minor}.{Patch}+{Build}` like https://semver.org/? Or should it just be `{Major}.{Minor}.{Patch}`?

Reply:
I don't have an opinion here, just followed the fo-dicom. But I do agree we should do similar to semver. I also see there is a restriction on number of characters which is 16. Sometimes our custom build will take more characters. I will stick to {Major}.{Minor}.{Patch}